CMakeLists.txt

```diff
@@ -51,13 +47,15 @@ target_include_directories(
     PUBLIC
     "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
     "$<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>"
+    "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>" # ?? although your exampleTargets.cmake has a INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include/example"
 )
 
 target_compile_features(example_example PUBLIC cxx_std_14)
 
 # ---- Install ----
-
-set(CMAKE_INSTALL_INCLUDEDIR "include/example" CACHE PATH "")
+# I would not mess with those variables, it confused me when your install looked like this:
+# include/example/example/example_export.h
```

Comment:
Or was this on purpose, because this way you can have multiple projects in the same install dir, and still keep the targets include dirs separated?
`include/<project>/<library>/headerfile.h` 🤔

Reply:
This is to avoid an issue outlined here: `https://github.com/Dobiasd/FunctionalPlus/issues/222`

> I would not mess with those variables

These variables are to be messed with, that's why they are cache variables. The reason why this is here is also explained in the git commit that added this.

Reply:
Wow, thats enlightning. I had this issue in my mind, and it makes total sense now. Many Libraries do that, OpenCV, Eigen etc. Will adopt accordingly, thanks!

Reply:
Followup question: Why not do that for all folders, `include`, `bin`, `lib` and optionally `share`?

Reply:
`bin`, `lib` and `share` aren't provided by a dependency to look for files in as directories. Files in those directories are referenced directly, so this problem can't arise there. The "problem" with the include directory is that the compiler is just pointed at a directory to look for files referenced by arbitrary `#include`s.